### ISSUE TITLE
"Overhauled bubble particle behavior" doesn't work with soul sand bubbles

### ISSUE THREAD

**Issue by Magicrafter13** [Tschipcraft/fabric_make_bubbles_pop_mod#3]
### Describe the bug

Bubbles aren't going around blocks (most of the time), as shown in the demonstration video.

### Steps to reproduce

Quilt 1.20.1 client, using latest version of mod on a vanilla(-compatible) server.

### Expected behavior

To look like the first video on this page: https://modrinth.com/mod/make_bubbles_pop

### Additional context

What it looks like on my client: https://matthewrease.net/simplescreenrecorder-2023-06-27_20.15.07.mp4 (ignore my drowning lol)

**Comment by Tschipcraft** [comment]
Sorry for the misconception, the overhauled bubble behavior only applies to regular bubbles particles and not the bubble particles emitting from soul sand or magma blocks. I did this, so that it doesn't appear like there is an upwards or downwards force when there is none.
In the demonstration video, I spawned the regular bubble particles by using the `/particles` command.

**Comment by Magicrafter13** [comment]
Ah, that's unfortunate. Would it be possible to implement with soul sand, or is that just a totally different type of system?

**Comment by Tschipcraft** [comment]
No, it's the same system. I would just have to copy some code from the regular bubble particle over to the soul sand bubble particle.
I'd however like to implement this as an optional feature, so I'll first have to figure out how to use an config api mod.

**Comment by Magicrafter13** [comment]
To throw my two cents into the ring, Mod Menu integration would probably be the best route for config.

**Comment by Tschipcraft** [comment]
Yes, that's for sure. I'm currently experimenting with [MidnightLib](https://modrinth.com/mod/midnightlib) and got basic configuration working.

**Comment by SabrCat** [comment]
just ran into this, would be cool to see this implemented still.

**Comment by Tschipcraft** [comment]
This is still in development, I plan on adding functionality for the bubbles to switch between their types (normal, soul sand, magma) based on their location before adding this. It will probably be in v0.4.0 or smth
